src/types/index.ts

```diff
@@ -867,3 +868,15 @@ export interface IAmountDecimalPlaces {
   highPrecision?: boolean;
   amount?: number;
 }
+
+export interface IAddressBookEntry {
+  id: string;
+  name: string;
+  address: AccountAddress;
+  isBookmarked: boolean;
+  protocol: Protocol;
+}
+
+export type IAddressBook = Record<string, IAddressBookEntry>;
+
+export type IAddressBookFilter = ObjectValues<typeof ADDRESS_BOOK_FILTERS> | Protocol | undefined;
```

Comment:
I'm not a big fan of mixing types together. I think better solution would be to split the filter into 2 variables: `isBookmarked` (type: bool) and `protocol` (type: Protocol | null).